code/game/turfs/simulated/floor/light_floor.dm

```diff
@@ -71,11 +90,24 @@
 		to_chat(user, "<span class='warning'>[src]'s light bulb appears to have burned out.</span>")
 
 /turf/simulated/floor/light/proc/toggle_light(light)
+	if(!on && !power_check())
+		visible_message("<span class='danger'>[src] doesn't react, it seems to be out of power.</span>")
+		return
+	var/area/A = get_area(src)
+	if(!A)
+		return
```

Comment:
this check isnt needed `power_check()` will fail if there is no area. (which it shouldnt do, if there is no area it should throw a runtime)